stock_inventory_draft_warning/wizard/stock_quant_wizard_views.xml

```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <record id="view_stock_quant_wizard_tree" model="ir.ui.view">
+        <field name="name">stock.quant.wizard.form</field>
+        <field name="model">stock.quant.wizard</field>
+        <field name="arch" type="xml">
+            <form string="Involved Products">
+                <group>
+                    <field name="warning_message" nolabel="1" readonly="1"></field>
+                </group>
+                <group>
+                    <label for="product_ids"/>
+                </group>
+                <group>
+                    <field name="product_ids" nolabel="1"/>
```

Comment:
Since this a `Many2many` field and no widget nor custom view was selected, default Odoo behavior in this case is showing a `tree` view for products, that is a bad choice IMO.
Please try `many2many_tags` widget instead.